lib/ui/compositing.dart

```diff
@@ -61,6 +61,8 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
       throw new ArgumentError('"matrix4" argument cannot be null');
     if (matrix4.length != 16)
       throw new ArgumentError('"matrix4" must have 16 entries.');
+    if (matrix4.any((double value) => value.isInfinite || value.isNaN))
+      throw new ArgumentError('"matrix4" cannot have infinite or NaN elements');
```

Comment:
Can we add this to the `_validateMatrix4` in `painting.dart`, and just call that from here?

Even though it's private, it's available here because this is all `part of` dart:ui

Reply:
This is outdated. We're now calling `_throwIfMarix4IsInvalid`.